### PR TITLE
modeling-signup pls check it for me

### PR DESCRIPTION
**Pull request by monleopa**

### CODE REVIEW

**Comment by monleopa** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find(params[:id])
```
Comment (huuphuc93) [inline app/controllers/users_controller.rb]: use find_by id: params[:id]

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def create
+    @user = User.new(user_params)
```
Comment (huuphuc93) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    if @user.save
+      flash[:success] = "Welcome to the Duc Anh!"
+      redirect_to @user
+    else
+      render 'new'
```
Comment (huuphuc93) [inline app/controllers/users_controller.rb]: 'new' => :new

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    end
+  end
+
+  private
+  def user_params
+    params.require(:user).permit(:name, :email, :password,
```
Comment (huuphuc93) [inline app/controllers/users_controller.rb]: remove () after permit

File: app/helpers/users_helper.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+module UsersHelper
+
+    # Returns the Gravatar for the given user.
```
Comment (huuphuc93) [inline app/helpers/users_helper.rb]: remove all comment and empty line

File: app/helpers/users_helper.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    def gravatar_for(user, size: 80)
```
Comment (huuphuc93) [inline app/helpers/users_helper.rb]: remove (), move 80 to file Setting(using gem config)

File: app/helpers/users_helper.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+      gravatar_id = Digest::MD5::hexdigest(user.email.downcase)
+      gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}"
+      image_tag(gravatar_url, alt: user.name, class: "gravatar")
```
Comment (huuphuc93) [inline app/helpers/users_helper.rb]: remove ()

File: app/helpers/users_helper.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+    end
+end
```
Comment (huuphuc93) [inline app/helpers/users_helper.rb]: add the break line at the end of this file

File: app/models/user.rb [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+    before_save { email.downcase! }
+    validates :name,  presence: true, length: { maximum: 50 }
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+    validates :email, presence: true, length: { maximum: 255 },
+                      format: { with: VALID_EMAIL_REGEX },
+                      uniqueness: { case_sensitive: false }
+
+    has_secure_password
+    validates :password, presence: true, length: { minimum: 6 }
+  end
```
Comment (huuphuc93) [inline app/models/user.rb]: add the break line at the end of this file

File: app/models/user.rb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+    before_save { email.downcase! }
```
Comment (huuphuc93) [inline app/models/user.rb:2]: { email.downcase! } => {email.downcase!}, same with all hash after

File: config/routes.rb [hunk 11]
```diff
@@ -1,6 +1,9 @@
...
-  root 'static_pages#home'  
-end
+  root "static_pages#home"
+  get  "/help",    to: "static_pages#help"
+  get  "/about",   to: "static_pages#about"
+  get  "/contact", to: "static_pages#contact"
+  get  "/signup",  to: "users#new"
+  post "/signup",  to: "users#create"
+  resources :users
+end
```
Comment (huuphuc93) [inline config/routes.rb]: add the break line after this file

File: test/fixtures/users.yml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (huuphuc93) [inline test/fixtures/users.yml]: remove this file

File: test/models/user_test.rb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (huuphuc93) [inline test/models/user_test.rb]: remove this file

File: app/views/users/show.html.erb [hunk 14]
```diff
@@ -0,0 +1,11 @@
+<% provide(:title, @user.name) %>
```
Comment (huuphuc93) [inline app/views/users/show.html.erb]: remove ()

File: app/views/users/new.html.erb [hunk 15]
```diff
@@ -0,0 +1,18 @@
+<% provide(:title, "Sign up") %>
```
Comment (huuphuc93) [inline app/views/users/new.html.erb]: remove ()

File: app/views/layouts/_header.html.erb [hunk 16]
```diff
@@ -1,6 +1,6 @@
 <header class="navbar navbar-fixed-top navbar-inverse">
     <div class="container">
```
Comment (huuphuc93) [inline app/views/layouts/_header.html.erb:2]: indent 2 spaces intead of 4

File: app/assets/stylesheets/custom.scss [hunk 17]
```diff
@@ -1,100 +1,173 @@
-@import "bootstrap-sprockets";
-@import "bootstrap";
+@import 'bootstrap-sprockets';
+@import 'bootstrap';
+@import 'color';
+
+@mixin box_sizing {
+	-moz-box-sizing:    border-box;
```
Comment (huuphuc93) [inline app/assets/stylesheets/custom.scss:6]: indent 2 spaces intead of 4 in this whole file

File: app/assets/stylesheets/color.scss [hunk 18]
```diff
@@ -0,0 +1,2 @@
+$gray-medium-light: #eaeaea;
+$light-gray: #bbb;
```
Comment (huuphuc93) [inline app/assets/stylesheets/color.scss]: add the break line at the end of this file

**Comment by huuphuc93** [comment]
comment
Combine 2 commits => 1 commit

**Comment by huuphuc93** [comment]
comment
Combine 2 commits => 1 commit

**Comment by huuphuc93** [comment]
commented
Combine 2 commits => 1 commit

**Comment by monleopa** [comment]
ready

**Comment by huuphuc93** [comment]
commented
fix all comment and combine 2 commits => 1 commit before ready

**Pull request merged**